### PR TITLE
Added/fixed filters, added bicleaner-hardrules as a filter

### PR DESCRIPTION
**Pull request by TudorMN**
- Added filters from OpusFilter
- Modified empty-train filters
- Fixed some minor issues in the previous git pull
- Added bicleaner-hardrules as an individual modifiable filter in `bicleaner_all_hardrules`
- Deleted `no_porn` filter, recommend using the one from the newly added bicleaner-hardrules filter

### CODE REVIEW

**Comment by jelmervdl** [comment]
@TudorMN @ZJaume I'm a bit confused as what happened to bicleaner_hardrules. I was under the impression that we would add them as separate filters, but implement all of them by just calling the existing python library.

Its a small difference, but quite a difference in the user experience. By splitting up the filter into separate filters you'd be able to search for them, and the interface will be easier to understand.

Implementation wise you can do something similar to what I did in #47 where all the json files call the same python script to execute the actual filter.

File: filters/avg_word_length.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+import sys
+from sys import stdin, stdout, stderr
+from unicodedata import category as cat
+import argparse
+
+def parse_user_args():
+    """Parse the arguments necessary for this filter"""
+    parser = argparse.ArgumentParser(description="Removes lines with average word lengths between specified min and max values")
+    parser.add_argument("--min-avg-length", default=20, type=int)
```
Comment (jelmervdl) [inline filters/avg_word_length.py:10]: Nitpick but default should be same as in the json: `default=2`

File: filters/no_porn.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    porn_tokenizer_lang1 = Tokenizer(command = command, l = args.src_lang)
+    porn_tokenizer_lang2 = Tokenizer(command = command, l = args.trg_lang)
+    
+    for line in stdin:
+        fields = line.strip().split('\t')
+        src = fields[-2].strip()
+        trg = fields[-1].strip()
+        
+        src_porn_pass = porn_tokenizer_lang1.tokenize(src.lower())
+        trg_porn_pass = porn_tokenizer_lang2.tokenize(trg.lower())
```
Comment (jelmervdl) [inline filters/no_porn.py:30]: This seems to tokenize. But it is missing the porn classification?

**Review by jelmervdl — CHANGES_REQUESTED** [review]

**Comment by XapaJIaMnu** [comment]
We already have bicleaner-hardrules. Are there things here that are waiting to be merged?

**Pull request closed without merging**